src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1100,10 +1108,35 @@ def create_or_replace_dynamic_table(
         if not is_sql_select_statement(child.queries[0].sql.lower().strip()):
             raise SnowparkClientExceptionMessages.PLAN_CREATE_DYNAMIC_TABLE_FROM_SELECT_ONLY()
 
+        if create_mode == DynamicTableCreateMode.OVERWRITE:
+            replace = True
+            if_not_exists = False
+        elif create_mode == DynamicTableCreateMode.ERROR_IF_EXISTS:
+            replace = False
+            if_not_exists = False
+        elif create_mode == DynamicTableCreateMode.IGNORE:
+            replace = False
+            if_not_exists = True
+        else:
+            # should never reach here
+            raise ValueError(f"Unknown create mode: {create_mode}")  # pragma: no cover
+
         child = child.replace_repeated_subquery_with_cte()
         return self.build(
             lambda x: create_or_replace_dynamic_table_statement(
-                name, warehouse, lag, comment, x
+                name,
+                warehouse,
+                lag,
+                comment,
+                replace,
+                if_not_exists,
+                refresh_mode,
+                initialize,
+                clustering_keys,
+                is_transient,
+                data_retention_time,
+                max_data_extension_time,
+                x,
```

Comment:
I suggest you use named arguments. Positional arguments are error prone when there are so many arguments.